bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/model/Layout.java

```diff
@@ -143,4 +152,45 @@ public String getLayoutView() {
             return "";
         }
     }
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Layout l = (Layout) o;
+        List<PositionDatum> pd = getPositionData();
+        List<PositionDatum> otherPd = l.getPositionData();
+
+        boolean positionDataEquals = false;
```

Comment:
Ah, then I maybe understand the other comments. It is used to detect if there is a change. Lets say you have the old layout data and the new layout data. If they are both null, then there is nothing to draw. It was nothing to draw before, and is nothing to draw now. If the old one was null, and the new one is not null, we want to draw the image. And if none of them are null, we need to check down the object structure.

Reply:
Yes the comments >//< where meant as inline Java comments that should show what I is happening at that point, so I can understand the intent or what is meant to happen. 
Maybe it is only me but I feel the code (even if it may be correct) is hard to understand in terms what its intention is. I tried to tell what I think it is doing which leads me to telling you,  I don't get it. 

Maybe some comments would help and adding a few tests in the test class that show what is meant to happen.